pkg/util/workloadspread/workloadspread.go

```diff
@@ -61,3 +123,453 @@ func VerifyGroupKind(ref *appsv1alpha1.TargetReference, expectedKind string, exp
+				return nil
+			}
+			klog.Errorf("get Pod(%s/%s) matched workloadSpread(%s) failed: %s", newPod.Namespace, newPod.Name, injectWS.Name, err.Error())
+			return err
+		}
+
+		return h.mutatingPod(matchedWS, newPod, injectWS, admissionv1beta1.Update)
+	}
+
+	return nil
+}
+
+func (h *Handler) HandlePodCreation(pod *corev1.Pod) error {
+	// filter out pods, include the following:
+	// 1. Deletion pod
+	// 2. Pod.Status.Phase = Succeeded or Failed
+	// 3. Pod.OwnerReference is nil
+	// 4. Pod.OwnerReference is not one of workloads, such as CloneSet, Deployment, ReplicaSet.
+	if !kubecontroller.IsPodActive(pod) {
+		return nil
+	}
+	ref := metav1.GetControllerOf(pod)
+	matched, err := matchReference(ref)
+	if err != nil || !matched {
+		return nil
+	}
+
+	var matchedWS *appsv1alpha1.WorkloadSpread
+	workloadSpreadList := &appsv1alpha1.WorkloadSpreadList{}
+	if err = h.Client.List(context.TODO(), workloadSpreadList, &client.ListOptions{Namespace: pod.Namespace}); err != nil {
+		return err
+	}
+	for _, ws := range workloadSpreadList.Items {
+		if ws.Spec.TargetReference == nil {
+			continue
+		}
+		// determine if the reference of workloadSpread and pod is equal
+		if h.isReferenceEqual(ws.Spec.TargetReference, ref, pod.Namespace) {
+			matchedWS = &ws
+			// pod has at most one matched workloadSpread
+			break
+		}
+	}
+	// not found matched workloadSpread
+	if matchedWS == nil {
+		return nil
+	}
+
+	return h.mutatingPod(matchedWS, pod, nil, admissionv1beta1.Create)
+}
+
+func (h *Handler) HandlePodDeletion(pod *corev1.Pod) error {
+	var injectWS *InjectWorkloadSpread
+	str, ok := pod.Annotations[MatchedWorkloadSpreadSubsetAnnotations]
+	if !ok || str == "" {
+		return nil
+	}
+	err := json.Unmarshal([]byte(str), &injectWS)
+	if err != nil {
+		klog.Errorf("parse Pod (%s/%s) annotations[%s]=%s failed: %s", pod.Namespace, pod.Name,
+			MatchedWorkloadSpreadSubsetAnnotations, str, err.Error())
+		return nil
+	}
+
+	// filter out pods, include the following:
+	// 1. Deletion pod
+	// 2. Pod.Status.Phase = Succeeded or Failed
+	// 3. Pod.OwnerReference is nil
+	ref := metav1.GetControllerOf(pod)
+	if injectWS == nil || !kubecontroller.IsPodActive(pod) || ref == nil {
+		return nil
+	}
+
+	matchedWS := &appsv1alpha1.WorkloadSpread{}
+	err = h.Client.Get(context.TODO(), client.ObjectKey{Namespace: pod.Namespace, Name: injectWS.Name}, matchedWS)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Warningf("Pod(%s/%s) matched WorkloadSpread(%s) Not Found", pod.Namespace, pod.Name, injectWS.Name)
+			return nil
+		}
+		klog.Errorf("get pod(%s/%s) matched workloadSpread(%s) failed: %s", pod.Namespace, pod.Name, injectWS.Name, err.Error())
+		return err
+	}
+
+	return h.mutatingPod(matchedWS, pod, injectWS, admissionv1beta1.Delete)
+}
+
+func (h *Handler) mutatingPod(matchedWS *appsv1alpha1.WorkloadSpread,
+	pod *corev1.Pod,
+	injectWS *InjectWorkloadSpread,
+	operation admissionv1beta1.Operation) error {
```

Comment:
Maybe just change `operation admissionv1beta1.Operation` to `isCreate bool`, it can also replace the `isInject` below.